### ISSUE TITLE
Excessive borders rendered in High Contrast mode for several components

### ISSUE THREAD

**Issue by mpriour** [Esri/calcite-components#4912]
### Summary

The [High Contrast Audit](https://github.com/Esri/calcite-components/wiki/High-Contrast-Audit) revealed that several components have excessive borders that make them overly visually busy in High Contrast mode. This was classified as a Severity 2 (Needs improvement) level issue in the audit.

The following components are effected by this issue:
* Accordion
* Combobox
* Dropdown
* Tree

### Actual Behavior

**Accordion**

<img height=360 src="https://user-images.githubusercontent.com/142636/149417142-457cf39e-39c5-48c8-bee9-cc91d9462544.png">

**Combobox**

<img height=360 src="https://user-images.githubusercontent.com/142636/149594457-919301e5-03a8-4cee-84e0-be2ad13be8f9.png">

**Dropdown**

<img height=200 src="https://user-images.githubusercontent.com/142636/149594708-4fb80b43-4a9c-4d02-a3a2-ddc69fff1c11.png">

**Tree**

<img height="300" src="https://user-images.githubusercontent.com/142636/149637369-67912db5-9128-46c3-b2ea-d650b054bff2.png">

### Expected Behavior

The effected components in High Contrast should be more similar to the normal rendering mode. There should not be lines and borders drawn where none are needed or present in the normal rendering mode.

### Reproduction Sample

https://codepen.io/mpriour/pen/ExEgrRp

### Reproduction Steps

Windows
1. Set Windows color theme to a high-contrast variant.
2. Open the reproduction sample (https://codepen.io/mpriour/pen/ExEgrRp) in any browser.
3. Observe the issue.

Mac
1. Open the reproduction sample (https://codepen.io/mpriour/pen/ExEgrRp) in Chrome or Edge.
2. Open Dev Tools.
3. Click on Elements tab.
4. Next to the console sub tab, there should be a "Rendering" tab (if not you will need to enable it)
5. On the Rendering tab choose Forced-colors: active from the "Emulate forced colors..." selector
6. Observe the issue.

### Reproduction Version

latest

**Comment by geospatialem** [comment]
Verified on `beta.68` 👍🏻 

`accordion`:
![verify-accordion-high-contrast](https://user-images.githubusercontent.com/5023024/179305850-e8ff7833-af89-470f-a528-b7aa6938a419.PNG)

`combobox`:
![verify-combobox-high-contrast](https://user-images.githubusercontent.com/5023024/179305851-192102fe-7a2e-42fd-b700-245c61138f58.PNG)

`dropdown`:
![verify-dropdown-high-contrast](https://user-images.githubusercontent.com/5023024/179305852-9468d33f-5e5e-4323-b446-88e59a6fc9b4.PNG)

`tree`:
![verify-tree-high-contrast](https://user-images.githubusercontent.com/5023024/179305856-3b3b7a44-f919-461a-82fc-8a79b3fbbc78.PNG)
